File: medium/cdr/cdr.js
function cons(a, b) {
  let pair = (f) => f(a, b);
  return pair;
}

function car(pair) {
  let returnFirst = (a, b) => a;
  return pair(returnFirst);
}

function cdr(pair) {
  let returnLast = (a, b) => b;
  return pair(returnLast);
}

/* Test */
console.log(car(cons('a', 'b'))); // 'a'
console.log(cdr(cons('a', 'b'))); // 'b'

File: medium/cdr/python/cdr.py
def cons(a, b):
    def pair(f):
        return f(a, b)
    return pair

def car(pair):
    """Return the first element of pair.
    >>> car(cons(1,2))
    1
    """
    def retrieveFirst(a, b):
        return a
    return pair(retrieveFirst)

def cdr(pair):
    """Return the last element of pair.
    >>> cdr(cons(1,2))
    2
    """
    def retrieveLast(a, b):
        return b
    return pair(retrieveLast)

print(car(cons(1,2))) # 1

print(cdr(cons(1,2))) # 2


# To run doctests, run the command "python cdr.py -v"
if __name__ == "__main__":
    import doctest
    doctest.testmod()

